webapp/src/components/AccordionLayout.tsx

```diff
@@ -0,0 +1,53 @@
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
+import {
+  Accordion,
+  AccordionDetails,
+  AccordionSummary,
+  Box,
+  Divider,
+  Typography,
+} from "@mui/material";
+import React from "react";
+import Description from "components/Description";
+
+type Props = {
+  name: string;
+  description: string;
+  link: string;
+  children: React.ReactNode;
+};
+
+const AccordionLayout: React.FC<Props> = ({
+  name,
+  description,
+  link,
+  children,
+}) => {
+  return (
+    <Accordion>
+      <AccordionSummary expandIcon={<ExpandMoreIcon />} id={name}>
+        <Typography
+          variant="body1"
+          fontWeight="bold"
+          sx={{
+            width: "20%",
+            flexShrink: 0,
+          }}
+        >
+          {name}
+        </Typography>
+        <Divider orientation="vertical" flexItem sx={{ marginX: 2 }} />
+        <Box display="flex" gap={2}>
+          <Typography variant="body1">{description}</Typography>
+          <Description link={link} />
+        </Box>
+      </AccordionSummary>
+      <AccordionDetails>
+        <Divider sx={{ marginBottom: 1 }} />
+        {children}
+      </AccordionDetails>
+    </Accordion>
+  );
+};
```

Comment:
You can save one indent here:
```suggestion
}) => (
  <Accordion>
    <AccordionSummary expandIcon={<ExpandMoreIcon />} id={name}>
      <Typography
        variant="body1"
        fontWeight="bold"
        sx={{
          width: "20%",
          flexShrink: 0,
        }}
      >
        {name}
      </Typography>
      <Divider orientation="vertical" flexItem sx={{ marginX: 2 }} />
      <Box display="flex" gap={2}>
        <Typography variant="body1">{description}</Typography>
        <Description link={link} />
      </Box>
    </AccordionSummary>
    <AccordionDetails>
      <Divider sx={{ marginBottom: 1 }} />
      {children}
    </AccordionDetails>
  </Accordion>
);
```

Reply:
Interesting! does specify the "return" keyword explicitly not required?

Reply:
Not if you don't user brackets for the body of the arrow function. In `Settings.tsx`, You used 21 such arrow functions with an implicit return.

Reply:
I resolved this in 8c9d99b1480c423a72af22923af90bbbb4683918